src/page/dataverse/dataverse.tsx

```diff
@@ -46,125 +46,144 @@ const filters: Filter[] = [
   },
   {
+    id: '11',
     type: 'dataset',
     label: 'RPG FRANCE 2020',
     description:
       'The graphical parcel register is a geographic database used as a reference for the instruction of Common Agricultural Policy (CAP) subsidies. The anonymized version distributed here as part of the public service for making reference data available contains the graphic data of parcels (since 2015) with their main crop. These data are produced by the Agency of Services and Payment (ASP) since 2007'
   },
   {
+    id: '12',
     type: 'service',
     label: 'Data Grouping',
     description: 'The Data Grouping tool allows you to organize identical data into groups.'
   },
   {
+    id: '13',
     type: 'dataset',
     label: 'AGRESTE 2020',
     description:
       'Annual agricultural statistics, consisting of the areas, yields and production of the French territory.'
   },
   {
+    id: '14',
     type: 'dataset',
     label: 'RPG AGRESTE 2020',
     description:
       'Data set resulting from the join between RPG and AGRESTE data. Obtained through a sequence of data processing using the OKP4 protocol.'
   },
+  { id: '15', type: 'service', label: 'Storage', description: 'Data storage service' },
   {
-    type: 'service',
-    label: 'Storage',
-    description: 'Data storage service'
-  },
-  {
+    id: '16',
     type: 'dataset',
     label: 'ADMIN EXPRESS COG 2020 REGION',
     description:
       'ADMIN EXPRESS allows cross-referencing with other data sources in order to build thematic representations of the territory according to an administrative granularity (commune, departmental district, department, region). ADMIN EXPRESS is available in a "COG" edition, in accordance with the official geographic code published each year by INSEE.'
   },
   {
+    id: '17',
     type: 'dataset',
     label: 'Reference table of NAF Rev2 codes',
     description:
       'The French Nomenclature of Activities (NAF) is a nomenclature of productive economic activities, mainly developed to facilitate the organization of economic and social information.'
   },
   {
+    id: '18',
     type: 'service',
     label: 'Data Join Geospatial',
     description: 'The Data Join Geospatial tool allows to join two georeferenced datasets.'
   },
   {
+    id: '19',
     type: 'dataset',
     label: 'BASE SIRENE - NAF',
     description:
       'This dataset is obtained using the join between the SIRENE database and the NAF code reference table to obtain information on the activity of establishments.'
   },
   {
+    id: '20',
     type: 'dataset',
     label: 'SIRENE database - agriculture, forestry and fishing',
     description:
       'This dataset refers to all the agricultural, forestry and fishing companies in France in activity. Obtained through a sequence of data processing using the OKP4 protocol.'
   }
 ]
 
+export const getResourceDetails = (id: string): DataverseItemDetails | undefined => {
```

Comment:
Time to use [Option](https://gcanti.github.io/fp-ts/modules/Option.ts.html) here.

```suggestion
export const getResourceDetails = (id: string): Option<DataverseItemDetails> => {
```

Reply:
powerful and complex !